staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -291,3 +295,32 @@ var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate
 		},
 	}, true, nil
 })
+
+var KubernetesPodIdentityUserConversion = UserConversionFunc(func(chain []*x509.Certificate) (*authenticator.Response, bool, error) {
+	podIdentity, err := kubernetesx509.PodIdentityFromCertificate(chain[0])
+	if err != nil {
+		return nil, false, fmt.Errorf("while extracting Kubernetes X.509 PodIdentity extension: %w", err)
+	}
+
+	if podIdentity == nil {
+		return nil, false, nil
+	}
+
+	fp := sha256.Sum256(chain[0].Raw)
+	id := "X509SHA256=" + hex.EncodeToString(fp[:])
+
+	return &authenticator.Response{
+		User: &user.DefaultInfo{
+			Name:   fmt.Sprintf("system:serviceaccount:%s:%s", podIdentity.Namespace, podIdentity.ServiceAccountName),
+			UID:    podIdentity.ServiceAccountUID,
+			Groups: serviceaccount.MakeGroupNames(podIdentity.Namespace),
+			Extra: map[string][]string{
+				"authentication.kubernetes.io/pod-name":  {podIdentity.PodName},
+				"authentication.kubernetes.io/pod-uid":   {podIdentity.PodUID},
+				"authentication.kubernetes.io/node-name": {podIdentity.NodeName},
+				"authentication.kubernetes.io/node-uid":  {podIdentity.NodeUID},
+				user.CredentialIDKey:                     {id},
+			},
+		},
+	}, true, nil
+})
```

Comment:
yeah, I think this can't be a stateless function, but has to curry in a pod getter function that can look up the identified pod to ensure it is still valid